assets/js/components/KeyMetrics/ConfirmSitePurposeChangeModal.js

```diff
@@ -115,14 +122,50 @@ function ConfirmSitePurposeChangeModal( {
 		setUIValues,
 	] );
 
-	const { saveUserInputSettings } = useDispatch( CORE_USER );
+	const userInputPurposeConversionEvents = useSelect( ( select ) => {
+		const isGA4Connected =
+			select( CORE_MODULES ).isModuleConnected( 'analytics-4' );
+
+		if ( ! isGA4Connected ) {
+			return [];
+		}
+
+		return select(
+			MODULES_ANALYTICS_4
+		).getUserInputPurposeConversionEvents();
+	} );
+
+	const {
+		saveUserInputSettings,
+		setKeyMetricsSetting,
+		saveKeyMetricsSettings,
+	} = useDispatch( CORE_USER );
 
 	const saveChanges = useCallback( async () => {
 		setIsSaving( true );
 		await saveUserInputSettings();
+
+		// Update 'includeConversionTailoredMetrics' key metrics setting with included
+		//conversion events, to mark that their respective metrics should be included in the
+		// list of tailored metrics and persist on the dashboard in case events are lost.
+		setKeyMetricsSetting(
+			'includeConversionTailoredMetrics',
+			userInputPurposeConversionEvents
```

Comment:
Since I replace the type of this setting, this is adapted to save proper value (events) on purpose change, so changes in this file are related to the changes needed in this PR